Source/WebCore/page/SecurityOriginData.h

```diff
@@ -90,19 +103,59 @@ struct SecurityOriginData {
     
     WEBCORE_EXPORT String toString() const;
 
-    URL toURL() const;
+    WEBCORE_EXPORT URL toURL() const;
 
 #if !LOG_DISABLED
     String debugString() const { return toString(); }
 #endif
+
+    static bool shouldTreatAsOpaqueOrigin(const URL&);
+
+    template<class Encoder> void encode(Encoder&) const;
+    template<class Decoder> static std::optional<SecurityOriginData> decode(Decoder&);
 };
 
+template<class Encoder>
+void SecurityOriginData::encode(Encoder& encoder) const
```

Comment:
Moving from WebCoreArgumentCoders.serialization.in to here is a step in the wrong direction.  This is what the ReturnEarlyIfTrue keyword is for.